packages/runtime/src/runtime/modal.tsx

```diff
@@ -4,106 +4,242 @@ import { createPortal } from "react-dom";
 import { Outlet } from "react-router-dom";
 import OpenInFullIcon from "@mui/icons-material/OpenInFull";
 import CloseFullscreenIcon from "@mui/icons-material/CloseFullscreen";
-import { endsWith } from "lodash-es";
+import { endsWith, last } from "lodash-es";
 import { CloseOutlined } from "@ant-design/icons";
 
 interface ModalProps {
   title?: string | React.ReactNode;
   maskClosable?: boolean;
+  hideCloseIcon?: boolean;
+  hideFullScreenIcon?: boolean;
+  onClose?: () => void;
   position?: "top" | "right" | "bottom" | "left";
   height?: string;
-  width?: string;
+  width?: string | number;
   margin?: string;
   padding?: string;
+  backgroundColor?: string;
+  horizontalOffset?: number;
+  verticalOffset?: number;
+  showShadow?: boolean;
 }
 
 interface ModalContextProps {
-  openModal: (content: React.ReactNode, options: ModalProps) => void;
-  closeModal: () => void;
+  openModal: (
+    content: React.ReactNode,
+    options: ModalProps,
+    isDialog?: boolean,
+  ) => void;
+  closeAllModals: () => void;
 }
 
 export const ModalContext = createContext<ModalContextProps | undefined>(
   undefined,
 );
 
 export const ModalWrapper: React.FC = () => {
-  const [visible, setVisible] = useState(false);
-  const [content, setContent] = useState<React.ReactNode>(null);
-  const [options, setOptions] = useState<ModalProps>({});
-  const [isFullScreen, setIsFullScreen] = useState(false);
-  const [modalWidth, setModalWidth] = useState(0);
-  const [modalHeight, setModalHeight] = useState(0);
-  const [key, setKey] = useState(0);
+  const [modalState, setModalState] = useState<{
+    visible: boolean[];
+    content: React.ReactNode[];
+    options: ModalProps[];
+    key: number[];
+    isDialog: boolean[];
+  }>({
+    visible: [],
+    content: [],
+    options: [],
+    key: [],
+    isDialog: [],
+  });
+  const [modalDimensions, setModalDimensions] = useState<{
+    modalWidth: number[];
+    modalHeight: number[];
+  }>({
+    modalWidth: [],
+    modalHeight: [],
+  });
```

Comment:
Why is this an object with individual array properties instead of an array of objects?